mobile/src/bundles/common/components/button/button.tsx

```diff
@@ -1,19 +1,122 @@
-import React from 'react';
+import { type IconDefinition } from '@fortawesome/free-solid-svg-icons';
+import {
+    type FontAwesomeIconStyle,
+    type Props,
+} from '@fortawesome/react-native-fontawesome';
+import { FontAwesomeIcon } from '@fortawesome/react-native-fontawesome';
+import React, { useCallback, useState } from 'react';
 import { Pressable } from 'react-native';
+import {
+    type PressableProps,
+    type StyleProp,
+    type ViewStyle,
+} from 'react-native';
 
 import { Text } from '~/bundles/common/components/components';
 
 import { styles } from './styles';
 
 type Properties = {
-    label: string;
-    onPress: () => void;
-};
+    label?: string;
+    isPrimary?: boolean;
+    labelStyles?: StyleProp<ViewStyle>;
+    icon?: IconDefinition;
+    iconProps?: Omit<Props, 'icon'>;
+} & PressableProps;
+
+type StyleRecord = Record<string, unknown>;
+
+const Button: React.FC<Properties> = ({
+    label = '',
+    labelStyles,
+    style,
+    isPrimary = true,
+    disabled = false,
+    icon,
+    iconProps,
+    ...properties
+}) => {
+    const [isPressed, setIsPressed] = useState(false);
```

Comment:
https://medium.com/timeless/a-simple-touch-feedback-with-the-new-pressable-component-b4d91c35456

Reply:
I've use this from the beginig, but as for me its too hard for understanding, when checks makes in jsx.

Reply:
There's no need in creating a separate useState for pressed when it's already returned from styles, you can pass this value to the function foe example

Reply:
noted, will fix

Reply:
Also, sometimes it's ok to add checks in jsx, e.x.:
`style={[styles.button, isError && styles.errorButton]}`
Maybe this option will be more readable for you

Reply:
fix